src/main/java/io/stargate/sgv2/jsonapi/config/feature/DataApiFeatureFlag.java

```diff
@@ -0,0 +1,56 @@
+package io.stargate.sgv2.jsonapi.config.feature;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+
+/**
+ * Set of "Feature Flags" that can be used to enable/disable certain features in the Data API.
+ * Enumeration defines the key used to introspect state of feature.
+ *
+ * <p>NOTE: although flag names are in upper case (like {@code TABLES}), the actual configuration
+ * uses lower-case names (like {@code tables}) (with proper prefix).
+ */
+public enum DataApiFeatureFlag {
+  /**
+   * API Tables feature flag: if enabled, the API will expose table-specific Namespace resource
+   * commands, and support Collection commands on Tables. If disabled, those operations will fail
+   * with {@link ErrorCode#TABLE_FEATURE_NOT_ENABLED}.
+   *
+   * <p>If no configuration specified (config or request), the feature will be Disabled.
+   */
+  TABLES("tables", false);
+
+  private final String featureName;
+
+  private final boolean enabledByDefault;
```

Comment:
The command above says "If no configuration specified (config or request), the feature will be Disabled."

consider: we should prob enforce this by default in the enum and remove the enabledByDefault setting. Concern would be that we end up with a mix of values and it gets confusing.

Reply:
Problem is that precedence is 3-layered:

1. Explicit value from FeaturesConfig (map keyed by feature)
2. Explicit value from request header if no (1)
3. Default if neither (1) or (2).

I originally thought of just leaving (3) to be `false` for simplicity; I can still do that and drop `enabledByDefault` as it is not strictly needed -- and can be added if/when needed (or something else that does what is needed).

Reply:
discussed - drop the enabledByDefault and always default to false. the mental model is these are named boolean values, that default to false.